pkg/utils/nodeutils.go

```diff
@@ -115,6 +121,79 @@ func IsMcfgPoolUsingKC(pool *mcfgv1.MachineConfigPool) (bool, string, error) {
 	return true, currentKCMC, nil
 }
 
+func AreKubeletConfigsRendered(pool *mcfgv1.MachineConfigPool, client runtimeclient.Client) (bool, error) {
+	// find out if pool is using a custom kubelet config
+	isUsingKC, currentKCMCName, err := IsMcfgPoolUsingKC(pool)
+	if err != nil {
+		return false, fmt.Errorf("failed to check if pool %s is using a custom kubelet config: %w", pool.Name, err)
+	}
+	if !isUsingKC || currentKCMCName == "" {
+		return true, nil
+	}
+	// if the pool is using a custom kubelet config, check if the kubelet config is rendered
+	kcmcfg := &mcfgv1.MachineConfig{}
+	err = client.Get(context.TODO(), types.NamespacedName{Name: currentKCMCName, Namespace: pool.Namespace}, kcmcfg)
+	if err != nil {
+		return false, fmt.Errorf("failed to get machine config %s: %w", currentKCMCName, err)
+	}
+
+	kc, err := GetKCFromMC(kcmcfg, client)
+	if err != nil {
+		return false, fmt.Errorf("failed to get kubelet config from machine config %s: %w", currentKCMCName, err)
+	}
+
+	var obj interface{}
+	err = json.Unmarshal(kcmcfg.Spec.Config.Raw, &obj)
+	if err != nil {
+		return false, fmt.Errorf("failed to unmarshal machine config %s: %w", currentKCMCName, err)
+	}
+
+	encodedKC, err := jsonpath.Get("storage.files[0].contents.source", obj)
+	if err != nil {
+		return false, fmt.Errorf("failed to get encoded kubelet config from machine config %s: %w", currentKCMCName, err)
+	}
+	encodedKCStr := encodedKC.(string)
+	if encodedKCStr == "" {
+		return false, fmt.Errorf("encoded kubeletconfig %s is empty", currentKCMCName)
+	}
+	mcPayloadPrefix := `data:text/plain,`
+	encodedKCStr = strings.TrimPrefix(encodedKCStr, mcPayloadPrefix)
```

Comment:
If we are expecting the prefix, shouldn't we be more defensive and error out if the prefix is not there?